versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/BaseCommitHelper.java

```diff
@@ -387,6 +387,17 @@ private void validateNamespacesExistForContentKeys(
     }
   }
 
+  static ObjId idForExpectedContent(StoreKey key, StoreIndex<CommitOp> headIndex) {
+    StoreIndexElement<CommitOp> expected = headIndex.get(key);
+    if (expected != null) {
+      CommitOp expectedContent = expected.content();
+      if (expectedContent.action().exists()) {
+        return expectedContent.value();
+      }
+    }
+    return null;
```

Comment:
do we use `@Nullable` on methods in the codebase? or return `Optional` ?

Reply:
both :D